### PR TITLE
[New recipe] qmodmaster-0.5.2

### PR DESCRIPTION
**Pull request by mazbrili**
A free Qt-based implementation of a ModBus master application

### CODE REVIEW

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 1]
```diff
@@ -0,0 +1,87 @@
+SUMMARY="A free Qt-based implementation of a ModBus master application"
+DESCRIPTION="A graphical user interface allows easy communication with ModBus \
+ RTU and TCP slaves. QModMaster also includes a bus monitor for examining all  \
+ traffic in the bus
+ QModMaster is based on libmodbus. for modbus  \
+ communication and on QsLog for logging.
+ Supports Haiku, Windows and Linux.
```
Comment (extrowerk) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:7]: I would remove this line.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     ModBus RTU/TCP Master
+     Monitor Bus activity [Raw Data and PDU details]
+"
+COPYRIGHT="2012-2020 Elbar"
+HOMEPAGE="https://sourceforge.net/projects/qmodmaster/"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://download.sourceforge.net/qmodmaster/qModMaster-code-0.5.2-3/qModMaster-code-$portVersion-3.zip
+CHECKSUM_SHA256="1754581e3059f14c31c584aa7abe1f1a602d1814f293823272b1158bbc179b81"
+SOURCE_FILENAME="qModMaster-code-$portVersion-3.zip"
```
Comment (extrowerk) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:19]: This line needed only if the filename doesnt contains any version info. Defining this means hp should download the file and then rename it to this. You defined the original filename here, so it is unnecessary.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+SOURCE_DIR="qModMaster-code-$portVersion"
+ADDITIONAL_FILES="qmodmaster.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:23]: Mark x86_gcc2 as broken:
```diff
- ARCHITECTURES="?x86_gcc2 x86 x86_64"
+ ARCHITECTURES="!x86_gcc2 x86 x86_64"
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+"
+COPYRIGHT="2012-2020 Elbar"
+HOMEPAGE="https://sourceforge.net/projects/qmodmaster/"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://download.sourceforge.net/qmodmaster/qModMaster-code-0.5.2-3/qModMaster-code-$portVersion-3.zip
+CHECKSUM_SHA256="1754581e3059f14c31c584aa7abe1f1a602d1814f293823272b1158bbc179b81"
+SOURCE_FILENAME="qModMaster-code-$portVersion-3.zip"
+SOURCE_DIR="qModMaster-code-$portVersion"
+ADDITIONAL_FILES="qmodmaster.rdef.in"
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:21]: You forgot to add the rdef file in the PR :)

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+-Features
+     ModBus RTU/TCP Master
+     Monitor Bus activity [Raw Data and PDU details]
+"
+COPYRIGHT="2012-2020 Elbar"
+HOMEPAGE="https://sourceforge.net/projects/qmodmaster/"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://download.sourceforge.net/qmodmaster/qModMaster-code-0.5.2-3/qModMaster-code-$portVersion-3.zip
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:17]: The provided link doesn't work for me? ```https://sourceforge.net/projects/qmodmaster/files/qModMaster-code-0.5.2-3.zip"``` does work however ...

**Comment by Begasus** [comment]
Missing libnetwork ...

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+CHECKSUM_SHA256="1754581e3059f14c31c584aa7abe1f1a602d1814f293823272b1158bbc179b81"
+SOURCE_FILENAME="qModMaster-code-$portVersion-3.zip"
+SOURCE_DIR="qModMaster-code-$portVersion"
+ADDITIONAL_FILES="qmodmaster.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+        SECONDARY_ARCHITECTURES="x86"
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:25]: Through the whole recipe you have been using spaces for indents, please use tab(s) where needed:
```diff
-         SECONDARY_ARCHITECTURES="x86"
+	SECONDARY_ARCHITECTURES="x86"
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        sed \
+                -e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
+                -e "s|@MAJOR@|$MAJOR|" \
+                -e "s|@MIDDLE@|$MIDDLE|" \
+                -e "s|@MINOR@|$MINOR|" \
+                -e "s|@LONG_INFO@|$LONG_INFO|" \
+                "$portDir"/additional-files/qmodmaster.rdef.in > qmodmaster.rdef
+
+        addResourcesToBinaries qmodmaster.rdef $appsDir/QModMaster
+        addAppDeskbarSymlink $appsDir/QModMaster
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:86]: This creates a broken link in Deskbar:
```diff
-         addAppDeskbarSymlink $appsDir/QModMaster
+	addAppDeskbarSymlink $appsDir/QModmaster/QModMaster
```

**Review by Begasus — CHANGES_REQUESTED** [review]

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 5 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:17]: sorry all, i have problem with my box..  .. and i make mistake .. seem i upload untested one.. i will try test this week

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 1 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:6]: ok.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 5 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: fixed url.. thanks

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 4 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:17]: @Begasus  thanks.. fixed

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 2 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:15]: thanks. done

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 3 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:20]: ok done.. thanks @Begasus

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 1 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: ok

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 6 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:22]: ok.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 7 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: ok

**Comment by mazbrili** [comment]
> Missing libnetwork ...
> **EDIT** add next line in qModMaster.pro to fix this:
> `haiku:LIBS += -lnetwork`

yeah. thanks @Begasus . patch sent

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 6 above)
Comment (Begasus) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:21]: Still seeing spaces in front of the lines, please revert to tabs

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 8]
```diff
@@ -0,0 +1,82 @@
+SUMMARY="A free Qt-based implementation of a ModBus master application"
+DESCRIPTION="A graphical user interface allows easy communication with ModBus \
+ RTU and TCP slaves.
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:3]: No space here:
```diff
-  RTU and TCP slaves.
+ RTU and TCP slaves.
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+-Features
+     ModBus RTU/TCP Master
+     Monitor Bus activity [ Raw Data and PDU details ]
+"
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:8]: This looks better in HaikuDepot (last double-qoute should be at the end of the last line):
```diff
--Features
-     ModBus RTU/TCP Master
-     Monitor Bus activity [ Raw Data and PDU details ]
-"
+Features
+* ModBus RTU/TCP Master
+* Monitor Bus activity [ Raw Data and PDU details ]"
```

**Comment by Begasus** [comment]
Build is OK on 32bit, only some cosmetic changes required :)

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 9 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: ok

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 8 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: ok

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 6 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:20]: seem ok now.
i'm curious.. how did you find it..? notepad++ not show the difference

Comment (Begasus) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:20]: Need to do another local pull with your changes, but I'm still seeing spaces here, you can check this by looking at the recipe in a browser and then select some text:
github

**Comment by Begasus** [comment]
@mazbrili github is still showing spaces in front of the lines?

**Comment by mazbrili** [comment]
> @mazbrili github is still showing spaces in front of the lines?

omg.. i'm still not seeing it..?

**Comment by humdingerb** [comment]
You're using spaces instead of tabs. To see that nicely, open the recipe in Koder and activate the "Special symbols" from the tool bar. :)

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 10]
```diff
@@ -0,0 +1,81 @@
+SUMMARY="A free Qt-based implementation of a ModBus master application"
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:1]: I'd move the "Qt-based" to the Description:
```suggestion
SUMMARY="A ModBus master application"
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+DESCRIPTION="A graphical user interface allows easy communication with ModBus \
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:2]:
```suggestion
DESCRIPTION="A Qt-based graphical user interface that allows easy \
communication with ModBus RTU and TCP slaves.
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+RTU and TCP slaves.
+
+Features
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:5]:
```suggestion
Features:
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+* ModBus RTU/TCP Master
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:6]:
```suggestion
* ModBus RTU/TCP master
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+* Monitor Bus activity [ Raw Data and PDU details ]"
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:7]:
```suggestion
* Monitor bus activity (raw data and PDU details)"
```

File: app-misc/qmodmaster/additional-files/qmodmaster.rdef.in [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+resource app_flags B_MULTIPLE_LAUNCH;
+
+resource app_version {
+	major  = @MAJOR@,
+	middle = @MIDDLE@,
+	minor  = @MINOR@,
+
+	variety = B_APPV_BETA,
+	internal = 0,
+	short_info = "QModMaster",
```
Comment (humdingerb) [inline app-misc/qmodmaster/additional-files/qmodmaster.rdef.in:11]:
```suggestion
	short_info = "A ModBus master application",
```

**Review by humdingerb — COMMENTED** [review]
Minor quibbles :)

**Comment by mazbrili** [comment]
> You're using spaces instead of tabs. To see that nicely, open the recipe in Koder and activate the "Special symbols" from the tool bar. :)

oh... thank you @humdingerb for your tips.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+ADDITIONAL_FILES="qmodmaster.rdef.in"
+PATCHES=qmodmaster-$portVersion.patchset
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	SECONDARY_ARCHITECTURES="x86"
+fi
+
+PROVIDES="
+        qmodmaster$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:25]: Still spaces here:
```diff
-         qmodmaster$secondaryArchSuffix = $portVersion
- +	qmodmaster$secondaryArchSuffix = $portVersion
```

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+	cp -rf ManModbus  $appsDir/QModMaster/
+	local APP_SIGNATURE="application/x-vnd.qmodmaster"
+	local MAJOR="`echo "$portVersion" | cut -d. -f1`"
+	local MIDDLE="`echo "$portVersion" | cut -d. -f2`"
+	local MINOR="`echo "$portVersion" | cut -d. -f3`"
+	local LONG_INFO="$SUMMARY"
+	sed \
+        	-e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
+	        -e "s|@MAJOR@|$MAJOR|" \
+	        -e "s|@MIDDLE@|$MIDDLE|" \
```
Comment (Begasus) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:74]: Still spaces here:
```diff
-        	-e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
-	        -e "s|@MAJOR@|$MAJOR|" \
-	        -e "s|@MIDDLE@|$MIDDLE|" \
+		-e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
+		-e "s|@MAJOR@|$MAJOR|" \
+		-e "s|@MIDDLE@|$MIDDLE|" \
```

**Review by Begasus — COMMENTED** [review]
Still seeing some spaces ...

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 17 above)
Comment (Begasus) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:74]: If you select the lines with your mouse here on github it shows the difference once you move to the new lines:

Schermafdruk van 2021-04-27 16-35-08

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 16 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: ok done

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 17 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: please check again

**Review by mazbrili — COMMENTED** [review]
check lgtm

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 18]
```diff
@@ -0,0 +1,81 @@
+SUMMARY="A ModBus master application"
+DESCRIPTION="A Qt-based graphical user interface that allows easy \
+communication with ModBus RTU and TCP slaves.
+RTU and TCP slaves.
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:4]: Left overs... :)

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Network$secondaryArchSuffix
+	lib:libQt5Xml$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:37]: Add an emtpy line above.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:lrelease$secondaryArchSuffix >= 5
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qmake$secondaryArchSuffix >= 5
+	cmd:unzip
+	"
+BUILD()
```
Comment (humdingerb) [inline app-misc/qmodmaster/qmodmaster-0.5.2.recipe:54]: Add an emtpy line above.

**Review by humdingerb — CHANGES_REQUESTED** [review]
Otherwise looks good.

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 18 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe]: oops... done

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 20 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:55]: ok

File: app-misc/qmodmaster/qmodmaster-0.5.2.recipe (hunk 19 above)
Comment (mazbrili) [reply app-misc/qmodmaster/qmodmaster-0.5.2.recipe:37]: ok

**Review by humdingerb — APPROVED** [review]

**Comment by mazbrili** [comment]
> Missing libnetwork ...
> **EDIT** add next line in qModMaster.pro to fix this:
> `haiku:LIBS += -lnetwork`

ok done

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks for sticking to it! :+1:

**Pull request merged**